src/lib/dnssd/minimal_mdns/ResponseBuilder.h

```diff
@@ -77,16 +88,15 @@ class ResponseBuilder
             return *this;
         }
 
-        chip::Encoding::BigEndian::BufferWriter out(mPacket->Start() + mPacket->DataLength(), mPacket->AvailableDataLength());
-
-        if (!record.Append(mHeader, type, out))
+        if (!record.Append(mHeader, type, mWriter))
         {
             mBuildOk = false;
         }
         else
         {
-            mPacket->SetDataLength(static_cast<uint16_t>(mPacket->DataLength() + out.Needed()));
+            mPacket->SetDataLength(static_cast<uint16_t>(mEndianOutput.Needed()));
```

Comment:
We can we be sure that `Fit()` was called beforehand? The `Needed()` accessor should,  in most cases, come after an explicit `Fit()` or be replaced by `if (writer.Fit(neededLen)) ... blah.write(xxx, neededLen);`